File: LuaLibs/src/Libs/EventHandle/CallOnceEventDispatch.lua
--[[
    @brief 一次事件分发，分发一次就清理
]]

local CallOnceEventDispatch = GlobalNS.Class(GlobalNS.EventDispatch)
local M = CallOnceEventDispatch

function M:ctor()

end

function dispatchEvent(dispatchObject)
    super.dispatchEvent(dispatchObject);
    selfclearEventHandle();
end

return M

File: PyLibs/src/Libs/EventHandle/CallOnceEventDispatch.py
# -*- coding: utf-8 -*-

from Libs.EventHandle.EventDispatch import EventDispatch;

'''
@brief 一次事件分发，分发一次就清理
'''

class CallOnceEventDispatch(EventDispatch):
    def __init__(self):
        super(CallOnceEventDispatch, self).__init__();
        
        self.mTypeId = "CallOnceEventDispatch";


    def dispatchEvent(self, dispatchObject):
        super(CallOnceEventDispatch, self).dispatchEvent(dispatchObject);
        self.clearEventHandle();


